autotest/gcore/overviewds.py

```diff
@@ -105,6 +122,8 @@ def test_overviewds_2():
     assert ds.GetMetadataItem('RPC', 'FOO') is None
     ds = None
 
+    gdal.GetDriverByName('GTiff').Delete('tmp/byte.tif')
```

Comment:
@rouault if we use pytest's `tmp_path` fixture, there's no need for manual cleanup of the temp files.